# Q1.

#  A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.


# Required Library

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from scipy import stats 
from scipy.stats import norm


In [2]:
#Load data set
data=pd.read_csv("Cutlets.csv")
data

Unit A  Unit B
0   6.8090  6.7703
1   6.4376  7.5093
2   6.9157  6.7300
3   7.3012  6.7878
4   7.4488  7.1522
5   7.3871  6.8110
6   6.8755  7.2212
7   7.0621  6.6606
8   6.6840  7.2402
9   6.8236  7.0503
10  7.3930  6.8810
11  7.5169  7.4059
12  6.9246  6.7652
13  6.9256  6.0380
14  6.5797  7.1581
15  6.8394  7.0240
16  6.5970  6.6672
17  7.2705  7.4314
18  7.2828  7.3070
19  7.3495  6.7478
20  6.9438  6.8889
21  7.1560  7.4220
22  6.5341  6.5217
23  7.2854  7.1688
24  6.9952  6.7594
25  6.8568  6.9399
26  7.2163  7.0133
27  6.6801  6.9182
28  6.9431  6.3346
29  7.0852  7.5459
30  6.7794  7.0992
31  7.2783  7.1180
32  7.1561  6.6965
33  7.3943  6.5780
34  6.9405  7.3875

In [3]:
data.head(10)

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522
5  7.3871  6.8110
6  6.8755  7.2212
7  7.0621  6.6606
8  6.6840  7.2402
9  6.8236  7.0503

In [4]:
data.describe()

Unit A     Unit B
count  35.000000  35.000000
mean    7.019091   6.964297
std     0.288408   0.343401
min     6.437600   6.038000
25%     6.831500   6.753600
50%     6.943800   6.939900
75%     7.280550   7.195000
max     7.516900   7.545900

In [5]:
X=data.iloc[:,0]
X.head(11)

0     6.8090
1     6.4376
2     6.9157
3     7.3012
4     7.4488
5     7.3871
6     6.8755
7     7.0621
8     6.6840
9     6.8236
10    7.3930
Name: Unit A, dtype: float64

In [6]:
Y=data.iloc[:,1]
Y.head(10)

0    6.7703
1    7.5093
2    6.7300
3    6.7878
4    7.1522
5    6.8110
6    7.2212
7    6.6606
8    7.2402
9    7.0503
Name: Unit B, dtype: float64

In [7]:
P_value = stats.ttest_ind(X,Y)
P_value

Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)

In [8]:
P_value[1]

0.4722394724599501

In [9]:
print(P_value[1] > 0.05) #compare p_value[1] with α = 0.05 (at 5% significance level)

True


# Conclusion
As p_value of 0.4722 is greater than α = 0.05 ; we accept null hypothesis.

Therefore, mean1 = mean2 which means there is no significant difference in diameters of the cutlets between the two units.

# Q2.

# A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.   Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% significance level.
 

# Required Library


In [10]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from scipy import stats 
from scipy.stats import norm

In [11]:
lab=pd.read_csv("LabTAT.csv")
lab

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0          185.35        165.53        176.70        166.13
1          170.49        185.91        198.45        160.79
2          192.77        194.92        201.23        185.18
3          177.33        183.00        199.61        176.42
4          193.41        169.57        204.63        152.60
..            ...           ...           ...           ...
115        178.49        170.66        193.80        172.68
116        176.08        183.98        215.25        177.64
117        202.48        174.54        203.99        170.27
118        182.40        197.18        194.52        150.87
119        182.09        215.17        221.49        162.21

[120 rows x 4 columns]

In [12]:
lab.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 120 entries, 0 to 119
Data columns (total 4 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Laboratory 1  120 non-null    float64
 1   Laboratory 2  120 non-null    float64
 2   Laboratory 3  120 non-null    float64
 3   Laboratory 4  120 non-null    float64
dtypes: float64(4)
memory usage: 3.9 KB


In [13]:
lab.describe()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
count    120.000000    120.000000    120.000000     120.00000
mean     178.361583    178.902917    199.913250     163.68275
std       13.173594     14.957114     16.539033      15.08508
min      138.300000    140.550000    159.690000     124.06000
25%      170.335000    168.025000    188.232500     154.05000
50%      178.530000    178.870000    199.805000     164.42500
75%      186.535000    189.112500    211.332500     172.88250
max      216.390000    217.860000    238.700000     205.18000

In [14]:
# ANOVA ftest : stats.f_oneway(col1,col2,col3,col4)
p_value = stats.f_oneway(lab.iloc[:,0],lab.iloc[:,1],lab.iloc[:,2],lab.iloc[:,3])
print("The p-value is ",p_value[1])

The p-value is  2.1156708949992414e-57


In [15]:
print(p_value[1] > 0.05) # compare it with α = 0.05

False


# Conclusion
Hence, the p-value for this dataset is less than 0.05 ie the 5% significance level.

Therefore, we can reject the null hypothesis which means at least one sample mean is different from the others.

Thus, there is difference between the TAT(turnaround time) of the different labs.

# Q3

# Required Library

In [16]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm

In [17]:
sale=pd.read_csv("BuyerRatio.csv")
sale

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [18]:
sale.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2 entries, 0 to 1
Data columns (total 5 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Observed Values  2 non-null      object
 1   East             2 non-null      int64 
 2   West             2 non-null      int64 
 3   North            2 non-null      int64 
 4   South            2 non-null      int64 
dtypes: int64(4), object(1)
memory usage: 208.0+ bytes


In [19]:
sale.describe()

East         West        North       South
count    2.000000     2.000000     2.000000    2.000000
mean   242.500000   832.500000   743.500000  410.000000
std    272.236111   976.514465   866.205807  480.832611
min     50.000000   142.000000   131.000000   70.000000
25%    146.250000   487.250000   437.250000  240.000000
50%    242.500000   832.500000   743.500000  410.000000
75%    338.750000  1177.750000  1049.750000  580.000000
max    435.000000  1523.000000  1356.000000  750.000000

Null hypothesis H0: male-female buyer ratios are same , they do not vary and are not related.

Alternate hypothesis Ha: male-female buyer ratios are not same, they vary and are significantly/somewhat related.

In [20]:
from scipy.stats import chi2_contingency 

In [21]:
obs_values = np.array([[50,142,131,70],[435,1523,1356,750]])
obs_values

array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]])

In [22]:
chi2_contingency(obs_values) #this gives output (chi2 stats value, p-value, df, expected observations)

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [23]:
chi2_contingency(obs_values)[0]

1.595945538661058

In [24]:
chi2_contingency(obs_values)[1]

0.6603094907091882

In [25]:
print(chi2_contingency(obs_values)[1] > 0.05) #this is the p-value compared to α = 0.05 (5% significance level)

True


# Conclusion
As the p-value (0.6603) is greater than α = 0.05, we accept the Null hypothesis.

Therefore, the male-female buyer ratios are same across regions and they are not related.

# Q4

# TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences


# Required Library

In [26]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm

In [27]:
Tele=pd.read_csv("Costomer+OrderForm.csv")
Tele

Phillippines   Indonesia       Malta       India
0     Error Free  Error Free   Defective  Error Free
1     Error Free  Error Free  Error Free   Defective
2     Error Free   Defective   Defective  Error Free
3     Error Free  Error Free  Error Free  Error Free
4     Error Free  Error Free   Defective  Error Free
..           ...         ...         ...         ...
295   Error Free  Error Free  Error Free  Error Free
296   Error Free  Error Free  Error Free  Error Free
297   Error Free  Error Free   Defective  Error Free
298   Error Free  Error Free  Error Free  Error Free
299   Error Free   Defective   Defective  Error Free

[300 rows x 4 columns]

In [28]:
Tele.describe()

Phillippines   Indonesia       Malta       India
count           300         300         300         300
unique            2           2           2           2
top      Error Free  Error Free  Error Free  Error Free
freq            271         267         269         280

In [29]:
Tele.Phillippines.value_counts()

Error Free    271
Defective      29
Name: Phillippines, dtype: int64

In [30]:
Tele.Indonesia.value_counts()

Error Free    267
Defective      33
Name: Indonesia, dtype: int64

In [31]:
Tele.Malta.value_counts()

Error Free    269
Defective      31
Name: Malta, dtype: int64

In [32]:
Tele.India.value_counts()

Error Free    280
Defective      20
Name: India, dtype: int64

In [33]:
#create contingency table
obs_values = np.array([[271,267,269,280],[29,33,31,20]])
obs_values

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [34]:
chi2_contingency(obs_values) # this gives output: (chi2 stats value, p-value, df, expected observations)

(3.858960685820355,
 0.2771020991233135,
 3,
 array([[271.75, 271.75, 271.75, 271.75],
        [ 28.25,  28.25,  28.25,  28.25]]))

In [35]:
print("The p_value is: ",chi2_contingency(obs_values)[1])

The p_value is:  0.2771020991233135


In [36]:
print(chi2_contingency(obs_values)[1] > 0.05) #this checks the obtained p-value is greater than 0.05 or not. 

True


# Conclusion
As the p-value (0.2777) is greater than α = 0.05, we accept the Null hypothesis.

Therefore, defective % does not vary across centers.